delfin/drivers/fujitsu/eternus/eternus_stor.py

```diff
@@ -352,3 +391,548 @@ def parse_alert(context, alert):
     @staticmethod
     def get_access_url():
         return 'https://{ip}'
+
+    def list_storage_host_initiators(self, ctx):
+        initiator_list = []
+        host_status = self.get_host_status()
+        host_fc_list = self.get_data(consts.GET_HOST_WWN_NAMES,
+                                     consts.HOST_TOTAL)
+        for host_fc in (host_fc_list or []):
+            if len(host_fc) < consts.HOST_TOTAL:
+                continue
+            fc_name = host_fc[consts.HOST_NAME_COUNT]
+            fc_wwn = host_fc[consts.HOST_WWN_COUNT]
+            state = host_status.get(fc_name)
+            initiator_item = {
```

Comment:
Duplicate code. The structure is the same but the parameter values are different. The method should be extracted and transferred as input parameters.